### PR TITLE
Adding new value to VA baseline name

### PR DESCRIPTION
**Pull request by yaakoviyun**
Adding new value to VA baseline name

### CODE REVIEW

**Comment by yaakoviyun** [comment]
@marstr  - can you merge?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json [hunk 1]
```diff
@@ -140,11 +141,12 @@
           {
             "name": "baselineName",
             "in": "path",
-            "description": "The name of the vulnerability assessment rule baseline.",
+            "description": "The name of the vulnerability assessment rule baseline (default implies a baseline on a database level rule and master for server level rule).",
```
[the remark this answers is not in the document]
Comment (yaakoviyun) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json]: currently we don't support a "server level" configuration. In the meantime, the user can set a server rule baseline in the context of a single database

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
These changes seem small enough that they don't warrant ARM review. @jaredmoo, @yaakoviyun, anything else that needs to happen before this is merged?

**Comment by jaredmoo** [comment]
Before your merge, let me chat with Yaakov and check that I agree with this design :)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json]: Ok, so if I set the master baseline for one database, that doesn't affect the master baseline for another database, is that right? When I look at my VA scan results, how do I know whether I need to update the current or master baseline?

**Comment by yaakoviyun** [comment]
Jared, you are correct.

With regard to choosing the baseline target, it can be found in the scan results file.
Each check (rule) result has a property that indicates whether it is a database level or server level .
Sent from Outlook
________________________________
מאת: Jared Moore <notifications@github.com>
‏‏נשלח: יום רביעי 30 מאי 2018 21:02:40
‏‏אל: Azure/azure-rest-api-specs
עותק: Yaakov Iyun; Mention
‏‏נושא: Re: [Azure/azure-rest-api-specs] Adding new value to VA baseline name (#3143)


@jaredmoo commented on this pull request.

________________________________

In specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F3143%23discussion_r191866154&data=02%7C01%7Cyaiyun%40microsoft.com%7Ca0abf34101404117dcc208d5c65789c9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636633001633703508&sdata=14a%2B9NR1c9uZT1TWpT%2F03OLYSwziFf3gawlZ%2BYU87Uw%3D&reserved=0>:

> @@ -140,11 +141,12 @@
           {
             "name": "baselineName",
             "in": "path",
-            "description": "The name of the vulnerability assessment rule baseline.",
+            "description": "The name of the vulnerability assessment rule baseline (default implies a baseline on a database level rule and master for server level rule).",


Ok, so if I set the master baseline for one database, that doesn't affect the master baseline for another database, is that right? When I look at my VA scan results, how do I know whether I need to update the current or master baseline?

**Review by jaredmoo — APPROVED** [review]
Sounds good

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json [hunk 2]
```diff
@@ -61,15 +61,16 @@
...
             "required": true,
             "type": "string",
             "enum": [
-              "default"
+              "Master",
+              "Default"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json]: I think this should be 'master' , 'default'. The C# code generator should capitalize the first letter for the enum name when it generated code.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json [hunk 3]
```diff
@@ -295,12 +298,44 @@
         }
       }
     },
+    "Resource": {
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json]: Please remove Resource & ProxyResource definitions

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json [hunk 4]
```diff
@@ -295,12 +298,44 @@
...
+      ],
+      "properties": {}
+    },
     "DatabaseVulnerabilityAssessmentRuleBaseline": {
       "description": "A database vulnerability assessment rule baseline.",
       "type": "object",
       "allOf": [
         {
-          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+          "$ref": "#/definitions/ProxyResource"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/databaseVulnerabilityAssessmentBaselines.json]: Please undo this, the common/v1 definition should be used

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Overall looks good, minor changes needed

**Review by jaredmoo — APPROVED** [review]

**Comment by yaakoviyun** [comment]
@marstr , can you merge?

**Pull request merged**